File: solutions/0205-isomorphic-strings/solution.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Character,Character>map=new HashMap<>();
        Set<Character>use=new HashSet<>();
        char[] arr=s.toCharArray();
        char[] brr=t.toCharArray();
        for(int i=0;i<s.length();i++)
        {
            char a=arr[i];
            char b=brr[i];
            if(map.containsKey(a))
            {
                if(map.get(a)!=b)
                    return false;
            }
            else
            {
                if (use.contains(b))
                    return false;
                map.put(a,b);
                use.add(b);
            }
        }
        return true;
    }
}

File: solutions/0205-isomorphic-strings/solution.c
bool isIsomorphic(char* s, char* t) {
    int a[256]={0},b[256]={0};
    for(int i=0;(s[i] && t[i]);i++)
    {
        if(a[s[i]]==0 && b[t[i]]==0)
        {
            a[s[i]]=t[i];
            b[t[i]]=s[i];
        }
        else if(a[s[i]]!=t[i] && b[t[i]]!=s[i])
            return false;
    }
    return strlen(s)==strlen(t);
}
